src/scythe-bidder/combination-selector.tsx

```diff
@@ -0,0 +1,266 @@
+            }}
+            css={{ display: "block" }}
+          >
+            {p.label}
+          </Button>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+function reducer(
+  state: FactionMatCombinations,
+  action: FMComboAction
+): FactionMatCombinations {
+  let result = cloneDeep(state);
+  switch (action.type) {
+    case FMComboActionType.TOGGLE:
+      result[action.faction][action.mat] = action.value;
+      return result;
+    case FMComboActionType.PRESET:
+      return { ...action.preset };
+    case FMComboActionType.FACTION_TOGGLE:
+      MATS_IFA.forEach((mat) => {
+        result[action.faction][mat] = action.value;
+      });
+      return result;
+    case FMComboActionType.MAT_TOGGLE:
+      FACTIONS_IFA.forEach((faction) => {
+        result[faction][action.mat] = action.value;
+      });
+      return result;
+  }
+}
+
+export default function CombinationSelectorFormItem({
+  value,
+  onChange,
+}: {
+  value: FactionMatCombinations;
+  onChange: (value: FactionMatCombinations) => void;
+}) {
+  const [combinations, dispatch] = React.useReducer(reducer, value);
+  const [visible, setVisible] = React.useState(false);
+  return (
+    <React.Fragment>
+      <Modal
+        visible={visible}
+        width={768}
+        onCancel={() => {
+          setVisible(false);
+        }}
+        onOk={() => {
+          onChange(combinations);
+          setVisible(false);
+        }}
+      >
+        <CombinationSelector
+          value={combinations}
+          onToggle={(f: Faction, m: Mat, v: boolean) => {
+            dispatch({
+              type: FMComboActionType.TOGGLE,
+              faction: f,
+              mat: m,
+              value: v,
+            });
+          }}
+          onFactionToggle={(f: Faction, v: boolean) => {
+            dispatch({
+              type: FMComboActionType.FACTION_TOGGLE,
+              faction: f,
+              value: v,
+            });
+          }}
+          onMatToggle={(m: Mat, v: boolean) => {
+            dispatch({
+              type: FMComboActionType.MAT_TOGGLE,
+              mat: m,
+              value: v,
+            });
+          }}
+          onPresetSelect={(v: FactionMatCombinations) => {
+            dispatch({ type: FMComboActionType.PRESET, preset: v });
+          }}
+        />
+      </Modal>
+      <Select
+        value={
+          PRESETS.find(({ value: v }) => isEqual(v, value))?.label ?? "custom"
+        }
+        onChange={(v) => {
+          const foundPreset = PRESETS.find(({ label }) => label === v);
+          if (foundPreset) {
+            onChange(foundPreset.value);
+          }
+        }}
+        onSelect={(v) => {
+          if (v === "custom") {
+            setVisible(true);
+          }
+        }}
+        css={{ width: "100px !important" }}
+      >
+        {PRESETS.filter(({ hide }) => !hide).map((preset) => (
+          <Select.Option key={preset.label} value={preset.label}>
+            {preset.label}
```

Comment:
To me it seems like Deselect all/Anything goes are useful options during custom but I wonder if they should be included here. Right now Deselect is not mapped due to hide but anything goes remains in the game settings options. This is not necessarily wrong just wanted to make sure that was the desired behavior.

Reply:
Yeah, I thought "Anything goes" might be something that someone wanted to select, so I chose not to hide it.